### PR TITLE
Make $__interval and $__interval_ms global variables available to panel title and graph yaxis label

### LINKED ISSUE #8341: [Feature Request] Have $__interval and $__interval_ms global variables available to Panel Title Field
The $__interval and $__interval_ms global variables don't seem to be available to the panel Title field. They do work in the metric query field. I am using Grafana 4.2.0 on linux and the graphite datasource.

### PR DESCRIPTION
**Pull request by TibboddiT**
**What this PR does / why we need it**:

`$__interval` and `$__interval_ms` cannot be used to dynamically update panel title or yaxis label (for graph panel), and this is an issue we hit again and again for a long time now.

See for references https://community.grafana.com/t/using-the-interval-or-interval-in-panel-title/5612 and #8341.

Tonight, I can't deal with it anymore. So here is an implementation.

**Which issue(s) this PR fixes**:

Fixes #8341

**Special notes for your reviewer**:

I tried the best i could to minimize changes. I'm not familiar with the current code base, and maybe there is a better place somewhere. Please tell me if you think so.

### CODE REVIEW

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 1]
```diff
@@ -330,6 +330,14 @@ export class PanelChrome extends PureComponent<Props, State> {
...
     });
 
+    if (data.request) {
+      if (!panel.scopedVars) {
+        panel.scopedVars = {};
+      }
+      panel.scopedVars.__interval = data.request.scopedVars.__interval;
+      panel.scopedVars.__interval_ms = data.request.scopedVars.__interval_ms;
+    }
+
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx]: not sure on doing this mutation on render is best, think inside onDataUpdate is better

File: public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx [hunk 2]
```diff
@@ -235,6 +235,14 @@ export class PanelChromeAngularUnconnected extends PureComponent<Props, State> {
       'panel-content--no-padding': plugin.noPadding,
     });
 
+    if (data.request) {
+      if (!panel.scopedVars) {
+        panel.scopedVars = {};
+      }
+      panel.scopedVars.__interval = data.request.scopedVars.__interval;
+      panel.scopedVars.__interval_ms = data.request.scopedVars.__interval_ms;
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx]: move this to onPanelDataUpdate instead

**Review by torkelo — COMMENTED** [review]
Think this is looking promising, just slight tweak :)

File: public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx (hunk 2 above)
Comment (TibboddiT) [reply public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx]: Done :)

**Comment by TibboddiT** [comment]
I'm curious... how can i get rid of the CI error ? It seems it is because the target branch is `master`... If so, what is the correct PR workflow ?

**Comment by torkelo** [comment]
@aknuds1 think we have a really tricky problem with recent CI changes, and the upload to GSC in the PR branches steps, that does not work for external PRs as they do not have access to security ENV variables

**Comment by TibboddiT** [comment]
OK, thanks. I see you set env vars related to GCP in [this script](https://github.com/grafana/grafana/blob/master/scripts/build/gpg-test-vars.sh), which, according to CI config, is executed when an external PR is made, https://github.com/grafana/grafana/blob/master/.circleci/config.yml#L365-L370 :

```
      - run:
          name: Package Grafana
          command: |
            if [[ -n "$CIRCLE_PR_NUMBER" ]]; then
              echo Using test GPG key pair since building a forked PR
              source scripts/build/gpg-test-vars.sh
            fi
```

The CI build fail at https://github.com/grafana/grafana/blob/master/.circleci/config.yml#L387-L394 :

```
          name: Upload packages to GCS
          command: |
            # Work around an issue with the gsutil installation
            export BOTO_CONFIG=/dev/null
            echo $GCP_GRAFANA_UPLOAD_KEY > /tmp/gcpkey.json
            ./google-cloud-sdk/bin/gcloud auth activate-service-account --key-file=/tmp/gcpkey.json
            ./google-cloud-sdk/bin/gsutil -m rsync -r dist gs://grafana-build-pipeline/oss/packages/${CIRCLE_WORKFLOW_ID}
```

Seems to me that you should set GCP_GRAFANA_UPLOAD_KEY to some json value in scripts/build/gpg-test-vars.sh.

**Comment by aknuds1** [comment]
@torkelo Ah oh no, that's something I didn't take into consideration. I'll have to try to fix this tomorrow then.

**Comment by aknuds1** [comment]
The CI issue is fixed, please merge with latest Grafana master in order to get the fixes.

**Comment by TibboddiT** [comment]
Done.

**Comment by torkelo** [comment]
Sorry for not merging, just a bit unsure how reliable mutating scopedVars like this

**Comment by TibboddiT** [comment]
No problem. What about adding a check for existence of `__interval` and `__interval_ms` on `scopedVars` ? Would it be OK ? The `scopedVars` reference is not modified, except when there is no `scopedVars` on `panel`. Or something more like https://github.com/grafana/grafana/pull/16556/files ?

**Pull request closed without merging**